lib/src/components/NxButton/__tests__/NxButton.test.tsx

```diff
@@ -5,82 +5,68 @@
 
-import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
 import NxButton from '../NxButton';
-import { TooltipContext } from '../../NxTooltip/NxTooltip';
 
 describe('NxButton', function() {
-  it('renders a button', function() {
-    const button = shallow(<NxButton />);
+  const quickRender = rtlRender(NxButton, {}),
+      renderEl = rtlRenderElement(NxButton, {});
 
-    expect(button).toHaveClassName('nx-btn');
+  it('renders a button', function() {
+    expect(quickRender().getByRole('button').tagName).toBe('BUTTON');
   });
 
-  it('renders a primary button', function() {
-    const button = shallow(<NxButton variant="primary">Primary Button</NxButton>);
+  it('sets ref on the button', function() {
+    const ref = React.createRef<HTMLButtonElement>(),
+        el = renderEl({ ref });
 
-    expect(button).toMatchSelector('button.nx-btn.nx-btn--primary');
-    expect(button).toHaveText('Primary Button');
+    expect(ref.current).toBe(el);
   });
 
-  it('renders a secondary button by default', function() {
-    const button = shallow(<NxButton>Secondary Button</NxButton>);
+  it('adds specified classNames to the element in addition to the defaults', function() {
+    const el = renderEl({ className: 'foo' }),
+        defaultEl = renderEl()!;
 
-    expect(button).toHaveClassName('.nx-btn--secondary');
+    expect(el).toHaveClass('foo');
+
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('renders an icon-only button', function() {
-    const button = shallow(<NxButton title="Check" variant="icon-only"><NxFontAwesomeIcon icon={faCheck}/></NxButton>);
+  it('passes additional attrs to the button', function() {
+    const btn = quickRender({ id: 'foo', lang: 'en-US' }).getByRole('button');
 
-    expect(button.find('button')).toMatchSelector('.nx-btn.nx-btn--icon-only');
+    expect(btn).toHaveAttribute('id', 'foo');
+    expect(btn).toHaveAttribute('lang', 'en-US');
   });
 
-  it('passes the disabled attribute through', function() {
-    const button = shallow(<NxButton variant="error" disabled>Disabled Button</NxButton>);
-
-    expect(button).toMatchSelector('button.nx-btn.nx-btn--error');
-    expect(button).toBeDisabled();
+  it('sets disabled on the button when the disabled prop is true', function() {
+    expect(quickRender().getByRole('button')).not.toBeDisabled();
+    expect(quickRender({ disabled: undefined }).getByRole('button')).not.toBeDisabled();
+    expect(quickRender({ disabled: false }).getByRole('button')).not.toBeDisabled();
+    expect(quickRender({ disabled: true }).getByRole('button')).toBeDisabled();
   });
 
   it('disabled by class button has aria-disabled true', function() {
-    const button = shallow(<NxButton className="disabled">Disabled Button</NxButton>);
-
-    expect(button).toHaveProp('aria-disabled', true);
+    expect(quickRender().getByRole('button')).not.toBeDisabled();
```

Comment:
I assume this was supposed to be
```suggestion
    expect(quickRender({ className: 'disabled' }).getByRole('button')).not.toBeDisabled();
```